core/factory.py

```diff
@@ -124,15 +124,35 @@ def is_member(self, uid: int, whitelist: List[int]) -> bool:
             # whitelist is a whitelist, we don't lock whitelisted users
             return True
         return False
-
-    def notify_about_membership(self, message) -> None:
+    
+    def is_blogger(self, message: Message) -> bool:
+        """
+        Checks if the user is writing as a channel
+        """
+        user = message.from_user
+        if user.first_name == 'Channel' and user.is_bot:
+            return True
+        return False
+    
+    def notify_about_membership(self, message, is_blogger: bool = False) -> None:
+        """
+        Notify the user that he/she is not a member,
+        if he/she is blogger also
+        """
         try:
-            self.send_message(message.from_user.id, self.strings.you_are_not_a_member)
+            if is_blogger:
+                text = self.strings.you_are_blogger
+                # delete sent message as a channel
+                self.delete_message(message.chat.id, message.message_id)
+            else:
+                text = self.strings.you_are_not_a_member
+                
+            self.send_message(message.from_user.id, text)
         except telebot.apihelper.ApiTelegramException:
             pass
         finally:
             # if user is not a member, using the bot is considered violation
-            self.set_context(message, 'violation', True)
+            None if is_blogger else self.set_context(message, 'violation', True)
```

Comment:
This is an expression, since we are not using its value, it is better not to use an expression here at all. This would be a better and more readable option:
```python
if is_blogger:
  self.set_context(message, 'violation', True)
```